internal/transport/http2_client.go

```diff
@@ -176,7 +176,8 @@ func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error
 	if networkType == "tcp" && useProxy {
 		return proxyDial(ctx, address, grpcUA)
 	}
-	return (&net.Dialer{}).DialContext(ctx, networkType, address)
+	// KeepAlive is set to a negative value to prevent Go's override of the TCP keepalive time and interval; retain the OS default.
```

Comment:
```suggestion
	// KeepAlive is set to a negative value to prevent Go's override of the TCP
	// keepalive time and interval; retain the OS default.
```